RUFAS/routines/animal/ration/ration_driver.py

```diff
@@ -309,64 +326,580 @@ class RationReporter:
-                     'potassium', 'N', 'EE', 'starch', 'TDN', 'DE', 'calcium']
+        nutrients = [
+            "DM",
+            "CP",
+            "ADF",
+            "NDF",
+            "lignin",
+            "ash",
+            "phosphorus",
+            "potassium",
+            "N",
+            "EE",
+            "starch",
+            "TDN",
+            "DE",
+            "calcium",
+        ]
 
         # feed nutrient amounts
         for key, val in ration.items():
-            nutrient_amount['dm'] += val
+            nutrient_amount["dm"] += val
             for nutr in nutrients:
                 # all values on a 100% dry matter basis
-                if nutr == 'DM':
-                    nutrient_amount['as_fed'] += val * (available_feeds[key][nutr] / 100)
-                elif nutr == 'N':
+                if nutr == "DM":
+                    nutrient_amount["as_fed"] += val * (available_feeds[key][nutr] / 100)
+                elif nutr == "N":
                     # [A.2.A.2]
-                    if key[:3] in ['121', '122', '155', '157']:
+                    if key[:3] in ["121", "122", "155", "157"]:
                         denom = 6.38
                     # [A.2.A.1]
                     else:
                         denom = 6.25
-                    nutrient_amount[nutr] += (available_feeds[key]['CP'] / (denom * 100)) * val
+                    nutrient_amount[nutr] += (available_feeds[key]["CP"] / (denom * 100)) * val
                 else:
                     nutrient_amount[nutr] += val * (available_feeds[key][nutr] / 100)
 
         # feed nutrient concentrations
-        dm_amount = nutrient_amount['dm']
+        dm_amount = nutrient_amount["dm"]
         if dm_amount == 0:
             dm_amount = 1
         for nutr in nutrients:
-            if nutr == 'DM':
-                nutrient_conc['dm'] = (nutrient_amount['as_fed'] / dm_amount) * 100
+            if nutr == "DM":
+                nutrient_conc["dm"] = (nutrient_amount["as_fed"] / dm_amount) * 100
             else:
                 # all values on a 100% dry matter basis
                 nutrient_conc[nutr] = (nutrient_amount[nutr] / dm_amount) * 100
         return nutrient_amount, nutrient_conc
 
+    @classmethod
+    def report_ration_supply(cls, ration: Dict, available_feeds: Dict, ration_report: Dict, body_weight: float) -> Dict:
+        """
+        Nutrient and energy supply of a formulated ration
+        Different from the report_ration method, since this takes into account digestibility and other factors
+
+        Parameters
+        ----------
+        ration : Dict
+            Dictionary of formulated ration, with keys as feed IDs, values as kg fed
+        available_feeds : Dict
+            feed.available_feeds
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
```

Comment:
```suggestion
            Animal body weight (kg).
```
This is the format we have been using in the Soil and Crop module, and it would be nice to make the modules consistent. However if this is the way the Animal module has units formatted in docstrings, then I think it would be better to keep it like this until we have time to go through all the modules and make them consistent.